Week2.py

```diff
@@ -0,0 +1,90 @@
+
+import unittest
+
+def find_triplets(string, target_sum):
+
+    ### this code works for any target sum value ####
+
+    if(string is None or ''):
+        return None
+
+    
+    size=len(string)
+    arr=[0]*size
+
+    ### converting input string into list of integer values ###
+    arr=map(int, string.split())
+    
+    arr.sort()
+
+    size=len(arr)
+
+    if size<3:
+        return None
+    
+    ### set containing required triplets ####
+    triplets_set=[]
+
+    ### pointer to traverse each variable ###
+    k=0
+
+    while(k<=(size-3)):
+
+        #### left pointer: moves towards right ####
+        l=k+1
+            
+        #### right pointer: moves towards left ###
+        r=size-1
+            
+        while(l<r):
+            
+            if((arr[l]+arr[r]+arr[k])==target_sum):
+                triplets_set.append((arr[k], arr[l], arr[r]))
+                l=l+1
+                r=r-1
+            
+            
+            while(arr[l]+arr[r]+arr[k]>target_sum):
+                r=r-1
+                if (r-1<=k):
```

Comment:
you could incorporate this if into the while condition to simplify this construct

Reply:
same for the other while loops

Reply:
Also, do you really need to go all the way to k or could you stop at l?